File: leet_1535_find_winner.ts
function getWinner(arr: number[], k: number): number {
    let hi:number = Math.max(...arr)
    let len:number = arr.length
    if (len <= k)
        return hi
    let count:number = 0
    let i = 0
    let res: number = arr[0]
    while (++i < len) {
        if (res < arr[i]) {
            count = 1
            res = arr[i]
        } else {
            count++
        }
        if (count == k)
            return res
    }
    // console.log(hi)
    return hi
};

File: leet_1535_find_winner.go
func getWinner(arr []int, k int) int {
    hi := arr[0]
    len := len(arr)
    i := 0
    for i < len {
        if hi < arr[i] {
            hi = arr[i]
        }
        i++
    }
    if k >= len {
        return hi
    }
    count := 0
    res := arr[0]
    i = 1
    for i < len {
        if res < arr[i] {
            res = arr[i]
            count = 1
        } else {
            count++
        }
        if count == k {
            return res
        }
        i++
    }
    return hi
}
